bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -126,6 +126,8 @@ private boolean isResultExpired(DiscoveryResult result, long now) {
 
     private Storage<DiscoveryResult> discoveryResultStorage;
 
+    private Map<DiscoveryResult, String> resultDiscovererMap = new HashMap<>();
```

Comment:
Would make the code simpler indeed.
Btw, as thingDiscovered() is not synchronized - shouldn't a ConcurrentHashMap be used here?